src/dosext/mfs/mfs.c

```diff
@@ -2035,8 +2035,49 @@ int find_file(char *fpath, struct stat * st, int root_len, int *doserrno)
 
   /* first see if the path exists as is */
   if (stat(fpath, st) == 0) {
-    Debug0((dbg_fd, "file exists as is\n"));
-    return (TRUE);
+
+#ifdef __linux__
+    if (file_on_fat(fpath) && S_ISREG(st->st_mode)) {
```

Comment:
You are very terse.
This "if" have no "else", which
means the non-fat case would
remain with SFN?

Reply:
The non fat case does what it always did, returns that the file exists on disk as is. As mentioned earlier, in the case that we try to access an SFN, it exists even though it doesn't appear in a normal POSIX directory listing. I see your point about appearing to treat ufs and vfat differently. The problem is that before this patch:
- find_file() with ufs SFN - as file doesn't exist on disk it would do all the processing afterwards including scan_dir() SFN-> LFN unmangling.
- find_file() with vfat SFN - returns true and no SFN->LFN conversion is done.

So whilst I think what I have is correct, perhaps it needs to be structured differently?